server/api/users/users.controller.ts

```diff
@@ -46,3 +54,34 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 
   res.json(user);
 };
+
+export const updateUserById: RequestHandler<{ id: string }, string, UpdatableFields> = async (req, res) => {
+
+  const acceptableFields = [
+    'bio',
+    'twitter_username',
+    'linkedin_url',
+    'github_username',
+    'website',
+  ];
+
+  const toUpdate: UpdatableFields = {};
+
+  acceptableFields.forEach((field) => {
+    if (field in req.body) {
+      toUpdate[field as keyof UpdatableFields] = req.body[field as keyof UpdatableFields];
+    }
+  });
+
+  const updated = await User.update(toUpdate, {
+    where: {
+      id: req.params.id,
+    },
+  });
+
+  if (updated[0] === 0) {
+    throw new NotFoundError('User not found');
+  }
+
+  res.json(`User of id ${req.params.id} updated.`);
```

Comment:
returning a string that basically amounts to "this action was successful" can be replaced by just sending a 200 response with no message (the client can deal with the message that gets surfaced in the UI).  however, we might have more use here to return the entire updated user object itself, so that the client can more easily update what it has stored.